reference-implementation/lib/readable-stream.js

```diff
@@ -187,6 +187,16 @@ class ReadableStream {
         rethrowAssertionErrorRejection(err);
       });
 
+      function waitForWritesToFinish() {
+        if (dest._state === 'writable' && WritableStreamCloseQueuedOrInFlight(dest) === false) {
+          // Another write may have started while we were waiting on this currentWrite, so we have to be sure to wait
+          // for that too.
+          const before = currentWrite;
```

Comment:
I find the name `before` a little unclear. I don't have strong feelings about it, but maybe `oldCurrentWrite` would be better?